hydromt_sfincs/sfincs.py

```diff
@@ -2852,7 +2866,16 @@ def read_subgrid(self):
                 self.logger.warning(f"sbgfile not found at {fn}")
                 return
 
-            self.reggrid.subgrid.load(file_name=fn, mask=self.mask)
+            # re-initialize subgrid (different variables for old/new version)
+            # TODO: come up with a better way to handle this
+            self.reggrid.subgrid = SubgridTableRegular()
+            self.subgrid = xr.Dataset()
```

Comment:
@DirkEilander you have suggestions here?

Since variables are slightly different between old/new methods, I think we always would like to start "clean" when we read a subgrid

Reply:
I think this is ok. But maybe I don't understand your concern. We can chat about it next week?

Reply:
Okay, then it's good, since it does what I was aiming for. I was just hoping that there would be more elegant ways to "re-initialize" these objects in the self.reggrid.subgrid.read_(binary) methods.